aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -120,5 +122,84 @@ Tensor prod(const Tensor &self, int64_t dim_, bool keepdim) {
 }
 
 // \DIM REDUCE ################################################################
+
+// MULTI DIM REDUCE ###########################################################
+
+template <Tensor (reduce_1)(const Tensor &, int64_t, bool)>
+inline Tensor reduce_multi(const Tensor &self, IntList dims_, bool keepdim) {
+  if (dims_.size() == 1) {
+    return reduce_1(self, dims_[0], keepdim);
+  }
+  if (dims_.size() == 0) {
+    return self;
+  }
+  size_t ndims = self.dim();
+  std::bitset<dim_bitset_size> seen = dim_list_to_bitset(dims_, ndims);
+  Tensor result = self;
+  for (size_t i = 0; i < dims_.size(); i++) {
+    size_t dim = maybe_wrap_dim(dims_[i], ndims);
+    result = reduce_1(result, dim, true);
```

Comment:
Wouldn't passing in `keepdim=false` here let us remove the next block? It seems safe to just collapse the dimensions, with some care taken to offset `dim` by `i`.

Reply:
Here,`dims_` is in the order the caller has given us, so we would have to keep track of the mapping input dims to reduced dims.
The crucial question is whether we want to let the user decide the order of reduction axes as the current code does. If we don't, we can reduce in order (as we squeeze now) with `keepdim=False`. If we do, the above seems to be more straightforward to me.